### DISCUSSION TITLE
How to replace a function.

### DISCUSSION THREAD

**Discussion by erastod** [vfsfitvnm/frida-il2cpp-bridge#294] [Q&A]
I'm trying to replace this code:

this code is the original from metadata:

public static int GetZhuoyueAttributeCount(int excellenceInfo)
{
if (excellenceInfo [= 0)
{
return 0;
}
int num = 0;
int num2 = 32;
for (int i = 0; i [ num2; i++)
{
if (Global.GetIntSomeBit(excellenceInfo, i) == 1)
{
num++;
}
}
return num;
}

and I want just to change the return value from
return num; to
return Math.Min(6,num); so the code must be like this

public static int GetZhuoyueAttributeCount(int excellenceInfo)
{
if (excellenceInfo [= 0)
{
return 0;
}
int num = 0;
int num2 = 32;
for (int i = 0; i [ num2; i++)
{
if (Global.GetIntSomeBit(excellenceInfo, i) == 1)
{
num++;
}
}
return Math.Min(6, num);
}

**Comment by ExternalAddress4401** [comment]
In that case you want something along the lines of this.

```
Il2Cpp.perform(() => {
    const assembly = Il2Cpp.domain.assembly('...').image;
    const klass = assembly.class('...');
    
    klass.method('GetZhuoyueAttributeCount').implementation = function(excellence: any) {
        const retValue = this.method('GetZhuoyueAttributeCount').invoke(excellence);
        return Math.min(6, retValue);
    }
})
```

You call the original method to get the return value and then return that as `Math.min` instead. You'll need to fill in the assembly and class based on what you get from [Il2Cpp.dump()](https://github.com/vfsfitvnm/frida-il2cpp-bridge/wiki/Snippets#dumping)
